src/renderer/qssWidget/js/qssMouseWidget.js

```diff
@@ -0,0 +1,715 @@
+                        onQssWidgetNotificationRequired: "{mouse}.events.onQssWidgetNotificationRequired"
+                    }
+                }
+            }
+        }
+    });
+
+
+    /**
+     * The QSS stepper widget for Mouse button
+     *
+     * Represents the quick set strip stepper widget. It is used for
+     * incrementing/decrementing a setting.
+    */
+    fluid.defaults("gpii.qssWidget.mouseSpeedStepper", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.selectorsTextRenderer"],
+
+        members: {
+            boundReachedHits: 0
+        },
+
+        specialErrorBoundHitTries: 2,
+        enableRichText: true,
+
+        model: {
+            messages: {
+                settingTitle: "{that}.model.setting.schema.title",
+                incrementButton: "Slower",
+                decrementButton: "Faster",
+                upperBoundError: "This is highest setting",
+                lowerBoundError: "This is lowest setting"
+            },
+            setting: {},
+
+            value: "{that}.model.setting.value"
+        },
+
+        styles: {
+            errorAnimation: "fl-qssStepperWidgetBtn-stepperErrorActivation",
+            warningAnimation: "fl-qssStepperWidgetBtn-stepperActivation"
+        },
+
+        selectors: {
+            settingTitle: ".flc-qssMouseSpeedStepperWidget-settingTitle",
+            indicators: ".flc-qssMouseSpeedStepperWidget-indicators",
+            stepperButton: ".flc-qssMouseSpeedStepperWidget-btn",
+            incButton: ".flc-qssMouseSpeedStepperWidget-incBtn",
+            decButton: ".flc-qssMouseSpeedStepperWidget-decBtn"
+        },
+
+        modelListeners: {
+            "setting.value": {
+                func: "{channelNotifier}.events.onQssWidgetSettingAltered.fire",
+                args: ["{that}.model.setting"],
+                includeSource: "fromWidget"
+            }
+        },
+
+        events: {
+            onLowerBoundReached: null,
+            onUpperBoundReached: null
+        },
+
+        listeners: {
+            onLowerBoundReached: {
+                funcName: "gpii.qssWidget.mouseSpeedStepper.handleBoundReached",
+                args: [
+                    "{that}",
+                    "{that}.model.messages.lowerBoundError"
+                ]
+            },
+            onUpperBoundReached: {
+                funcName: "gpii.qssWidget.mouseSpeedStepper.handleBoundReached",
+                args: [
+                    "{that}",
+                    "{that}.model.messages.upperBoundError"
+                ]
+            },
+
+            "onCreate.attachAnimationClearer": {
+                funcName: "gpii.qssWidget.mouseSpeedStepper.clearElementsAnimation",
+                args: [
+                    "{that}.dom.stepperButton",
+                    "{that}.options.styles"
+                ]
+            }
+        },
+
+        invokers: {
+            activateIncBtn: {
```

Comment:
We've now got a great opportunity here to remove all this code which is now duplicated in 3 places - 
i) here
ii) qssStepperWidget.js
iii) qssVolumeWidget.js

Please consolidate into a shared grade

Reply:
This is true, we will look into it, but probably will create a tech debt type task specifically for this, because its a bit outside the scope of this PR (it affects few other buttons). We will discuss it tomorrow after today's meeting.